### DISCUSSION TITLE
I got duplicated compile problems with MS C/C++ extension

### DISCUSSION THREAD

**Discussion by wkjung** [microsoft/vscode-cmake-tools#2571] [Q&A] [answered]
When I invoke `>CMake: build` with MS C/C++ extension (the official one), I got duplicated problems in the problem plane for the opened windows/files (for not-opened files, C/C++ errors does not show up). For example one error is shown as C/C++ error while the other is shown as GC error. The example below shows it:

![image](https://user-images.githubusercontent.com/39818948/170434599-37beafc4-8d87-4583-8265-1a1e37f0f5db.png)

I set the configuration like ` "configurationProvider": "ms-vscode.cmake-tools"`. Is this intended or I'm experiencing somewhat of a bug? How do I change this behavior?

**Comment by bobbrow** [comment, marked as answer]
> I set the configuration like "configurationProvider": "ms-vscode.cmake-tools". Is this intended or I'm experiencing somewhat of a bug? How do I change this behavior?

This is currently the expected behavior.  Both extensions are reporting the same problem because they are aware of it and want to let you know.  Neither extension turns off errors based on the existence of another extension that might also report the error.  You have a couple of options for how to manage the problems report though:
1. The Problems panel has a text box in the top-right corner that lets you hide errors based on an expression.  You can type "C/C++" to see just the ones from the C++ extension, or "GCC" to see the ones from the build.
2. If you want to keep the build errors, but not the IntelliSense errors, you can add this setting to your settings file: `"C_Cpp.errorSquiggles": "Disabled"`
3. If you want to keep the IntelliSense errors, but not the build errors, unfortunately that's broken right now. We need to fix our setting to hide the problems: #1932
